Flaky test: TestAttestAgent

```
--- FAIL: TestAttestAgent (0.64s)
    --- FAIL: TestAttestAgent/attest_with_join_token_only_works_once (0.03s)
        logs.go:18: 
            	Error Trace:	/home/runner/work/spire/spire/pkg/server/api/agent/v1/logs.go:18
            	            				/home/runner/work/spire/spire/pkg/server/api/agent/v1/service_test.go:3011
            	            				/home/runner/work/spire/spire/pkg/server/api/agent/v1/service_test.go:3058
            	Error:      	Not equal: 
            	            	expected: []spiretest.LogEntry{spiretest.LogEntry{Level:0x2, Message:"Invalid argument: failed to attest: join token does not exist or has already been used", Data:logrus.Fields{"node_attestor_type":"join_token"}}, spiretest.LogEntry{Level:0x4, Message:"API accessed", Data:logrus.Fields{"node_attestor_type":"join_token", "status":"error", "status_code":"InvalidArgument", "status_message":"failed to attest: join token does not exist or has already been used", "type":"audit"}}}
            	            	actual  : []spiretest.LogEntry{spiretest.LogEntry{Level:0x2, Message:"Invalid argument: failed to attest: join token does not exist or has already been used", Data:logrus.Fields{"node_attestor_type":"join_token"}}, spiretest.LogEntry{Level:0x4, Message:"API accessed", Data:logrus.Fields{"node_attestor_type":"join_token", "status":"error", "status_code":"InvalidArgument", "status_message":"failed to attest: join token does not exist or has already been used", "type":"audit"}}, spiretest.LogEntry{Level:0x4, Message:"API accessed", Data:logrus.Fields{"agent_id":"spiffe://example.org/spire/agent/join_token/test_token", "node_attestor_type":"join_token", "status":"success", "type":"audit"}}}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,2 +1,2 @@
            	            	-([]spiretest.LogEntry) (len=2) {
            	            	+([]spiretest.LogEntry) (len=3) {
            	            	  (spiretest.LogEntry) {
            	            	@@ -18,2 +18,12 @@
            	            	   }
            	            	+ },
            	            	+ (spiretest.LogEntry) {
            	            	+  Level: (logrus.Level) 4,
            	            	+  Message: (string) (len=12) "API accessed",
            	            	+  Data: (logrus.Fields) (len=4) {
            	            	+   (string) (len=8) "agent_id": (string) (len=54) "spiffe://example.org/spire/agent/join_token/test_token",
            	            	+   (string) (len=18) "node_attestor_type": (string) (len=10) "join_token",
            	            	+   (string) (len=6) "status": (string) (len=7) "success",
            	            	+   (string) (len=4) "type": (string) (len=5) "audit"
            	            	+  }
            	            	  }
            	Test:       	TestAttestAgent/attest_with_join_token_only_works_once
            	Messages:   	unexpected logs
FAIL
```

I have seen this flavor of test failure frequently in recent runs:
```
--- FAIL: TestAttestAgent (0.87s)
    --- FAIL: TestAttestAgent/attest_with_join_token_only_works_once (0.03s)
        logs.go:28: 
            	Error Trace:	/home/runner/work/spire/spire/test/spiretest/logs.go:28
            	            				/home/runner/work/spire/spire/pkg/server/api/agent/v1/service_test.go:3070
            	            				/home/runner/work/spire/spire/pkg/server/api/agent/v1/service_test.go:3120
            	Error:      	Not equal: 
            	            	expected: []spiretest.LogEntry{spiretest.LogEntry{Level:0x2, Message:"Invalid argument: failed to attest: join token does not exist or has already been used", Data:logrus.Fields{"node_attestor_type":"join_token"}}, spiretest.LogEntry{Level:0x4, Message:"API accessed", Data:logrus.Fields{"node_attestor_type":"join_token", "status":"error", "status_code":"InvalidArgument", "status_message":"failed to attest: join token does not exist or has already been used", "type":"audit"}}}
            	            	actual  : []spiretest.LogEntry{spiretest.LogEntry{Level:0x4, Message:"API accessed", Data:logrus.Fields{"node_attestor_type":"join_token", "status":"error", "status_code":"InvalidArgument", "status_message":"failed to attest: join token does not exist or has already been used", "type":"audit"}}, spiretest.LogEntry{Level:0x4, Message:"API accessed", Data:logrus.Fields{"agent_id":"spiffe://example.org/spire/agent/join_token/test_token", "node_attestor_type":"join_token", "status":"success", "type":"audit"}}}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,9 +1,2 @@
            	            	 ([]spiretest.LogEntry) (len=2) {
            	            	- (spiretest.LogEntry) {
            	            	-  Level: (logrus.Level) 2,
            	            	-  Message: (string) (len=86) "Invalid argument: failed to attest: join token does not exist or has already been used",
            	            	-  Data: (logrus.Fields) (len=1) {
            	            	-   (string) (len=18) "node_attestor_type": (string) (len=10) "join_token"
            	            	-  }
            	            	- },
            	            	  (spiretest.LogEntry) {
            	            	@@ -18,2 +11,12 @@
            	            	   }
            	            	+ },
            	            	+ (spiretest.LogEntry) {
            	            	+  Level: (logrus.Level) 4,
            	            	+  Message: (string) (len=12) "API accessed",
            	            	+  Data: (logrus.Fields) (len=4) {
            	            	+   (string) (len=8) "agent_id": (string) (len=54) "spiffe://example.org/spire/agent/join_token/test_token",
            	            	+   (string) (len=18) "node_attestor_type": (string) (len=10) "join_token",
            	            	+   (string) (len=6) "status": (string) (len=7) "success",
            	            	+   (string) (len=4) "type": (string) (len=5) "audit"
            	            	+  }
            	            	  }
            	Test:       	TestAttestAgent/attest_with_join_token_only_works_once
            	Messages:   	unexpected logs
FAIL
FAIL	github.com/spiffe/spire/pkg/server/api/agent/v1	2.029s
```

--- a/pkg/server/api/agent/v1/service_test.go
+++ b/pkg/server/api/agent/v1/service_test.go
@@ -1484,12 +1484,14 @@ func TestGetAgent(t *testing.T) {
 		},
 		{
 			name: "success - with mask",
-			req: &agentv1.GetAgentRequest{Id: &types.SPIFFEID{TrustDomain: "example.org", Path: "/spire/agent/agent-1"},
+			req: &agentv1.GetAgentRequest{
+				Id: &types.SPIFFEID{TrustDomain: "example.org", Path: "/spire/agent/agent-1"},
 				OutputMask: &types.AgentMask{
 					AttestationType:      true,
 					X509SvidExpiresAt:    true,
 					X509SvidSerialNumber: true,
-				}},
+				},
+			},
 			agent: &types.Agent{
 				Id:                   expectedAgents[agent1].Id,
 				AttestationType:      expectedAgents[agent1].AttestationType,
@@ -1510,8 +1512,10 @@ func TestGetAgent(t *testing.T) {
 		},
 		{
 			name: "success - with all false mask",
-			req: &agentv1.GetAgentRequest{Id: &types.SPIFFEID{TrustDomain: "example.org", Path: "/spire/agent/agent-1"},
-				OutputMask: &types.AgentMask{}},
+			req: &agentv1.GetAgentRequest{
+				Id:         &types.SPIFFEID{TrustDomain: "example.org", Path: "/spire/agent/agent-1"},
+				OutputMask: &types.AgentMask{},
+			},
 			agent: &types.Agent{
 				Id: expectedAgents[agent1].Id,
 			},
@@ -1832,7 +1836,8 @@ func TestRenewAgent(t *testing.T) {
 					Level:   logrus.ErrorLevel,
 					Message: "Invalid argument: failed to parse CSR",
 					Data: logrus.Fields{
-						logrus.ErrorKey: malformedError.Error()},
+						logrus.ErrorKey: malformedError.Error(),
+					},
 				},
 				{
 					Level:   logrus.InfoLevel,
@@ -2211,7 +2216,6 @@ func TestAttestAgent(t *testing.T) {
 		rateLimiterErr    error
 		dsError           []error
 	}{
-
 		{
 			name:       "empty request",
 			request:    &agentv1.AttestAgentRequest{},
@@ -2517,6 +2521,25 @@ func TestAttestAgent(t *testing.T) {
 			expectMsg:  "failed to attest: join token does not exist or has already been used",
 			expectLogs: []spiretest.LogEntry{
 				{
+					Level:   logrus.InfoLevel,
+					Message: "Agent attestation request completed",
+					Data: logrus.Fields{
+						telemetry.Address:          "",
+						telemetry.AgentID:          "spiffe://example.org/spire/agent/join_token/test_token",
+						telemetry.NodeAttestorType: "join_token",
+					},
+				},
+				{
+					Level:   logrus.InfoLevel,
+					Message: "API accessed",
+					Data: logrus.Fields{
+						telemetry.AgentID:          "spiffe://example.org/spire/agent/join_token/test_token",
+						telemetry.Status:           "success",
+						telemetry.Type:             "audit",
+						telemetry.NodeAttestorType: "join_token",
+					},
+				},
+				{
 					Level:   logrus.ErrorLevel,
 					Message: "Invalid argument: failed to attest: join token does not exist or has already been used",
 					Data: logrus.Fields{
@@ -2576,6 +2599,25 @@ func TestAttestAgent(t *testing.T) {
 				{Type: "test_type", Value: "result"},
 			},
 			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.InfoLevel,
+					Message: "Agent attestation request completed",
+					Data: logrus.Fields{
+						telemetry.AgentID:          "spiffe://example.org/spire/agent/test_type/id_with_result",
+						telemetry.NodeAttestorType: "test_type",
+						telemetry.Address:          "",
+					},
+				},
+				{
+					Level:   logrus.InfoLevel,
+					Message: "API accessed",
+					Data: logrus.Fields{
+						telemetry.Status:           "success",
+						telemetry.Type:             "audit",
+						telemetry.AgentID:          "spiffe://example.org/spire/agent/test_type/id_with_result",
+						telemetry.NodeAttestorType: "test_type",
+					},
+				},
 				{
 					Level:   logrus.InfoLevel,
 					Message: "Agent attestation request completed",
@@ -3065,12 +3107,8 @@ func TestAttestAgent(t *testing.T) {
 						e.Data[telemetry.Address] = ""
 					}
 				}
-				if tt.retry {
-					// Prevent cases where audit logs from previous calls are pushed after log is reset
-					spiretest.AssertLastLogs(t, test.logHook.AllEntries(), tt.expectLogs)
-				} else {
-					spiretest.AssertLogs(t, test.logHook.AllEntries(), tt.expectLogs)
-				}
+
+				spiretest.AssertLogsAnyOrder(t, test.logHook.AllEntries(), tt.expectLogs)
 			}()
 
 			ctx, cancel := context.WithCancel(context.Background())
@@ -3097,8 +3135,6 @@ func TestAttestAgent(t *testing.T) {
 				// make sure that the first request went well
 				require.NoError(t, err)
 				require.NotNil(t, result)
-				// clear entries from the previous run
-				test.logHook.Reset()
 
 				// attest once more
 				stream, err = test.client.AttestAgent(ctx)
@@ -3368,7 +3404,8 @@ func attest(t *testing.T, stream agentv1.Agent_AttestAgentClient, request *agent
 			request = &agentv1.AttestAgentRequest{
 				Step: &agentv1.AttestAgentRequest_ChallengeResponse{
 					ChallengeResponse: challenge,
-				}}
+				},
+			}
 
 			continue
 		}

